src/formatDate.js

```diff
@@ -8,7 +8,44 @@
  * @returns {string}
  */
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  const slicedDate = date.split(fromFormat[3]);
+  const newSeparator = toFormat[3];
+
+  const dayIndex = fromFormat.indexOf('DD');
+  const monthIndex = fromFormat.indexOf('MM');
+  const yearIndex = Math.abs(dayIndex + monthIndex - 3);
+
+  const newDayIndex = toFormat.indexOf('DD');
+  const newMonthIndex = toFormat.indexOf('MM');
+  const newYearIndex = Math.abs(newDayIndex + newMonthIndex - 3);
+
+  let formatedYear = slicedDate[yearIndex];
```

Comment:
The variable 'formatedYear' is misspelled. It should be 'formattedYear' to maintain proper English spelling.